csds-client/client/client.go

```diff
@@ -0,0 +1,228 @@
+package client
+
+import (
+	csdspb "github.com/envoyproxy/go-control-plane/envoy/service/status/v2"
+	envoy_type_matcher "github.com/envoyproxy/go-control-plane/envoy/type/matcher"
+
+	"context"
+	"crypto/x509"
+	"errors"
+	"flag"
+	"fmt"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/oauth"
+	"google.golang.org/grpc/metadata"
+	"time"
+)
+
+type Flag struct {
+	uri             string
+	platform        string
+	authnMode       string
+	apiVersion      string
+	requestFile     string
+	requestYaml     string
+	jwt             string
+	configFile      string
+	monitorInterval time.Duration
+}
+
+type Client struct {
+	cc         *grpc.ClientConn
+	csdsClient csdspb.ClientStatusDiscoveryServiceClient
+
+	nm   []*envoy_type_matcher.NodeMatcher
+	md   metadata.MD
+	info Flag
+}
+
+// ParseFLags parses flags to info
+func ParseFlags() Flag {
+	uriPtr := flag.String("service_uri", "trafficdirector.googleapis.com:443", "the uri of the service to connect to")
+	platformPtr := flag.String("cloud_platform", "gcp", "the cloud platform (e.g. gcp, aws,  ...)")
+	authnModePtr := flag.String("authn_mode", "auto", "the method to use for authentication (e.g. auto, jwt, ...)")
+	apiVersionPtr := flag.String("api_version", "v2", "which xds api major version  to use (e.g. v2, v3 ...)")
+	requestFilePtr := flag.String("request_file", "", "yaml file that defines the csds request")
+	requestYamlPtr := flag.String("request_yaml", "", "yaml string that defines the csds request")
```

Comment:
This was suggested by @htuch to behave like the [command options](https://www.envoyproxy.io/docs/envoy/latest/operations/cli#cmdoption-config-yaml) of Envoy.

> I think this could come on stdin, via a file or via flag with string. No need to force this to only be a file. If you take a look at how Envoy's bootstrap works, it can consume from a file a proto, but also can consume the bootstrap (or some override) from CLI. The idea is to rely on proto merge semantics (MergeFrom).